Manifold/Type.swift

```diff
@@ -45,19 +133,48 @@ public enum Type: Hashable {
 
 	public var hashValue: Int {
 		return analysis(
+			ifBase: { $0.hashValue },
 			ifVariable: { $0.hashValue },
 			ifFunction: { $0.hashValue ^ $1.hashValue },
 			ifUniversal: { $0.hashValue ^ $1.hashValue }
 		)
 	}
+
+
+	// MARK: Printable
+
+	public var description: String {
+		return describe()
+	}
+
+	private func describe(_ boundVariables: Set<Manifold.Variable> = []) -> String {
+		let bound = "α"
+		let free = "τ"
+		return analysis(
+			ifBase: { $0.description },
+			ifVariable: { (boundVariables.contains($0) ? bound : free) + $0.value.subscriptedDescription },
+			ifFunction: { t1, t2 in
+				let parameter = t1.describe(boundVariables) |> { t1.quantifiedType?.isFunction ?? t1.isFunction ? "(\($0))" : $0 }
+				return "\(parameter) → \(t2.describe(boundVariables))"
+			},
+			ifUniversal: {
+				let variables = lazy($0)
+					.map { bound + $0.value.subscriptedDescription }
+					|> sorted
+					|> (join <| ",")
+				return "∀{\(variables)}.\($1.describe(boundVariables.union($0)))"
+			})
+	}
```

Comment:
`Type` is now `Printable` for ease of debugging.

This `describe` function is employed to enable us to print quantified variables distinct from free ones. Thus, `αᵢ` is a quantified variable which is otherwise equal to a free variable `τᵢ`.